File: graphql-dgs-codegen-core/src/integTest/java/com/netflix/graphql/dgs/codegen/java/testcases/inputs/inputWithDefaultBigDecimal/expected/types/OrderFilter.java
package com.netflix.graphql.dgs.codegen.java.testcases.inputs.inputWithDefaultBigDecimal.expected.types;

import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.math.BigDecimal;
import java.util.Objects;

public class OrderFilter {
  private BigDecimal min = new BigDecimal("1.1");

  private BigDecimal avg = new BigDecimal(1.12);

  private BigDecimal max = new BigDecimal(3.14E+19);

  public OrderFilter() {
  }

  public OrderFilter(BigDecimal min, BigDecimal avg, BigDecimal max) {
    this.min = min;
    this.avg = avg;
    this.max = max;
  }

  public BigDecimal getMin() {
    return min;
  }

  public void setMin(BigDecimal min) {
    this.min = min;
  }

  public BigDecimal getAvg() {
    return avg;
  }

  public void setAvg(BigDecimal avg) {
    this.avg = avg;
  }

  public BigDecimal getMax() {
    return max;
  }

  public void setMax(BigDecimal max) {
    this.max = max;
  }

  @Override
  public String toString() {
    return "OrderFilter{min='" + min + "', avg='" + avg + "', max='" + max + "'}";
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    OrderFilter that = (OrderFilter) o;
    return Objects.equals(min, that.min) &&
        Objects.equals(avg, that.avg) &&
        Objects.equals(max, that.max);
  }

  @Override
  public int hashCode() {
    return Objects.hash(min, avg, max);
  }

  public static Builder newBuilder() {
    return new Builder();
  }

  public static class Builder {
    private BigDecimal min = new BigDecimal("1.1");

    private BigDecimal avg = new BigDecimal(1.12);

    private BigDecimal max = new BigDecimal(3.14E+19);

    public OrderFilter build() {
      OrderFilter result = new OrderFilter();
      result.min = this.min;
      result.avg = this.avg;
      result.max = this.max;
      return result;
    }

    public Builder min(BigDecimal min) {
      this.min = min;
      return this;
    }

    public Builder avg(BigDecimal avg) {
      this.avg = avg;
      return this;
    }

    public Builder max(BigDecimal max) {
      this.max = max;
      return this;
    }
  }
}

File: graphql-dgs-codegen-core/src/integTest/kotlin/com/netflix/graphql/dgs/codegen/cases/inputWithDefaultBigDecimal/expected/types/OrderFilter.kt
package com.netflix.graphql.dgs.codegen.cases.inputWithDefaultBigDecimal.expected.types

import com.fasterxml.jackson.`annotation`.JsonCreator
import com.fasterxml.jackson.`annotation`.JsonProperty
import com.netflix.graphql.dgs.codegen.GraphQLInput
import java.math.BigDecimal
import kotlin.Any
import kotlin.Pair
import kotlin.String
import kotlin.collections.List

public class OrderFilter @JsonCreator constructor(
  @JsonProperty("min")
  public val min: BigDecimal = default<OrderFilter, BigDecimal>("min", java.math.BigDecimal("1.1")),
  @JsonProperty("avg")
  public val avg: BigDecimal = default<OrderFilter, BigDecimal>("avg", java.math.BigDecimal(1.12)),
  @JsonProperty("max")
  public val max: BigDecimal = default<OrderFilter, BigDecimal>("max",
      java.math.BigDecimal(3.14E+19)),
) : GraphQLInput() {
  override fun fields(): List<Pair<String, Any?>> = listOf("min" to min, "avg" to avg, "max" to max)
}
